qan-api2/models/metrics.go

```diff
@@ -1120,26 +1116,63 @@ func (m *Metrics) GetSelectedQueryMetadata(ctx context.Context, periodStartFromS
 	}
 	defer rows.Close()
 
+	metadata := make(map[string][]string)
+	columnNames, err := rows.Columns()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get column names")
+	}
+	// during scanning it will skip duplicate values for same name
 	for rows.Next() {
-		err = rows.Scan(
-			&res.ServiceName,
-			&res.Database,
-			&res.Schema,
-			&res.Username,
-			&res.ReplicationSet,
-			&res.Cluster,
-			&res.ServiceType,
-			&res.ServiceId,
-			&res.Environment,
-			&res.NodeId,
-			&res.NodeName,
-			&res.NodeType)
+		row := make([]any, len(columnNames))
+		for i := range row {
+			row[i] = &row[i]
+		}
```

Comment:
It seems we have some problem here.

Reply:
Without that I cannot read it into map. Any ideas how to handle this better?

Reply:
Aha, it seems that I got what you're trying to do. Try this, it should be easier to understand:
```suggestion
		row := make([]any, len(columnNames))
		for i := range columnNames {
			row[i] = new(sql.RawBytes)
		}
```

Reply:
Or `new(any)`, probably will work as well

Reply:
Nice thanks. I will test it next week right after national holidays.

Reply:
In the end I used new(string).